### PR TITLE
Galaxy server config updates

### PR DESCRIPTION
**Pull request by bcoca**
also handle some corner cases with exceptions

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ansible-galaxy

### CODE REVIEW

File: lib/ansible/cli/galaxy.py
Comment (s-hertel) [inline lib/ansible/cli/galaxy.py:218]: We could also add common arguments for username and password.

cc @newswangerd since you've requested this a couple times :smile:

edit: No need to hold up this PR to add that, though, would be nice if --timeout made it into 2.13.

File: lib/ansible/cli/galaxy.py
Comment (s-hertel) [inline lib/ansible/cli/galaxy.py]: This fails in CI with a yaml.representer.RepresenterError because we use  safe_dump
```suggestion
    'token': {'default': None},
```

File: lib/ansible/cli/galaxy.py
Comment (s-hertel) [inline lib/ansible/cli/galaxy.py]:
```suggestion
            token_val = server_options['token'] or NoTokenSentinel
```

File: lib/ansible/cli/galaxy.py
Comment (s-hertel) [inline lib/ansible/cli/galaxy.py:218]: New option needs a changelog.

**Review by s-hertel — COMMENTED** [review]
@bcoca could you rebase this?

File: lib/ansible/cli/galaxy.py
Comment (newswangerd) [reply lib/ansible/cli/galaxy.py:218]: @s-hertel that would be really nice for vanilla pulp ansible installations, which don't currently have token auth.

Comment (bcoca) [reply lib/ansible/cli/galaxy.py:218]: i'll leave that to a followup

**Review by s-hertel — APPROVED** [review]

**Pull request merged**
